src/Exception/UnsupportedAlgorithmException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace SimpleSAML\XMLSecurity\Exception;
+
+/**
+ * Class UnsupportedAlgorithmException
+ *
+ * This exception is thrown when we can't handle a given algorithm
+ *
+ * @package simplesamlphp/xml-security
+ */
+class UnsupportedAlgorithmException extends RuntimeException
```

Comment:
Should `BlacklistedAlgorithmException` extend from this one? In the end it's the same situation, so I'm not even sure we need to distinguish between the two of them...